### PR TITLE
Source lrg target

### LINKED ISSUE #7161: Slasher doesnt handle attestation edge case well
# 🐞 Bug Report

### Description

Slasher doesn't handle unexpected cases like source > target well. Underflow, then ignored, while still potentially slashable.

### PR DESCRIPTION
**Pull request by shayzluf**
Feature

**What does this PR do? Why is it needed?**
Attestation with higher source than target epoch could be potentially found in the network
This pr makes an improvement in handling these edge cases on: 
- update spans
- detection
- sig update

**Which issues(s) does this PR fix?**

Fixes #7161

### CODE REVIEW

File: slasher/detection/attestations/spanner.go [hunk 1]
```diff
@@ -64,11 +68,21 @@ func (s *SpanDetector) DetectSlashingsForAttestation(
 	defer traceSpan.End()
 	sourceEpoch := att.Data.Source.Epoch
 	targetEpoch := att.Data.Target.Epoch
-	if (targetEpoch - sourceEpoch) > params.BeaconConfig().WeakSubjectivityPeriod {
+	dis := targetEpoch - sourceEpoch
+
+	if sourceEpoch > targetEpoch { //Prevent underflow and handle source > target slashable cases.
```
Comment (nisdas) [inline slasher/detection/attestations/spanner.go:73]: why flip source and target epoch here ?

Comment (shayzluf) [reply slasher/detection/attestations/spanner.go:73]: spans are recorded at source epoch. we need to get the right spans for detection to work right

File: slasher/detection/attestations/spanner_test.go [hunk 2]
```diff
@@ -490,7 +513,9 @@ func TestSpanDetector_DetectSlashingsForAttestation_Surround(t *testing.T) {
 				},
 				AttestingIndices: []uint64{0},
 			}
+			t.Log("before detect")
```
Comment (rauljordan) [inline slasher/detection/attestations/spanner_test.go:516]:
```suggestion
```

File: slasher/detection/attestations/spanner_test.go [hunk 3]
```diff
@@ -490,7 +513,9 @@ func TestSpanDetector_DetectSlashingsForAttestation_Surround(t *testing.T) {
...
 			res, err := sd.DetectSlashingsForAttestation(ctx, att)
+			t.Logf("after detect %v", res)
```
Comment (rauljordan) [inline slasher/detection/attestations/spanner_test.go:518]:
```suggestion
			t.Logf("after detect %v", res)
```

**Review by rauljordan — APPROVED** [review]

**Pull request merged**